atest/header_filter.robot

```diff
@@ -24,11 +62,12 @@ Setup up server for test
 
 Define a protocol with string field in header
     New protocol    StringInHeader
-    Chars     *     string_field 
+    Uint      1     integer_field
+    Chars     *     string_field
```

Comment:
Maybe add a terminator for this field? We could then try sending two messages and then receiving the correct one with the filter?